File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> sS;
        int end = 0;

        for(int i = 0; i < tokens.size();i++){
            if(tokens[i] == "/"){
                int x = sS.top();sS.pop();
                int y = (sS.top() / x); sS.pop();
                sS.push(y);
            } else if(tokens[i] == "*"){
                int x = sS.top();sS.pop();
                int y = (sS.top() * x); sS.pop();
                sS.push(y);
            } else if(tokens[i] == "+"){
                int x = sS.top();sS.pop();
                int y = (sS.top() + x); sS.pop();
                sS.push(y);
            } else if(tokens[i] == "-"){
                int x = sS.top();sS.pop();
                int y = (sS.top() - x); sS.pop();
                sS.push(y);
            } else {sS.push(stoi(tokens[i]));}
        }
        return sS.top();
    }
};

File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = ''.join(c.lower() for c in s if c.isalnum())
        stack = []
        midpoint = len(s) // 2
        for i in range(len(s)):
            if i < midpoint:
                stack.append(s[i])
            elif i >= (midpoint + (1 if len(s) % 2 != 0 else 0)):
                if s[i] != stack.pop():
                    return False
        return True